src/software/new_geom/util/intersection.cpp

```diff
@@ -0,0 +1,189 @@
+#include "software/new_geom/util/intersection.h"
+
+/**
+ * Computes the point of intersection between two lines.
+ * Note: this computes the intersection of two lines, not line segments.
+ *
+ * See:
+ * https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection#Given_two_points_on_each_line
+ *
+ * @pre the lines are not parallel
```

Comment:
Why can't this function handle parallel lines? Wouldn't that be why we return an optional?

Reply:
Left over from old geom body. I removed it and added a note that overlapping lines arent considered as intersecting